circuit/src/commonMain/kotlin/com/slack/circuit/EventListener.kt

```diff
@@ -8,11 +8,67 @@ package com.slack.circuit
  */
 public interface EventListener {
 
-  /** Called when there is a new [state] returned by the [Presenter]. */
+  /** Called just before creating a [Presenter] for a given [screen]. */
+  public fun onBeforeCreatePresenter(
+    screen: Screen,
+    navigator: Navigator,
+    context: CircuitContext
+  ) {}
+
+  /**
+   * Called just after creating a [presenter] for a given [screen]. The presenter may be null if
+   * none was found.
+   */
+  public fun onAfterCreatePresenter(
+    screen: Screen,
+    navigator: Navigator,
+    presenter: Presenter<*>?,
+    context: CircuitContext
+  ) {}
+
+  /** Called just before creating a [Ui] for a given [screen]. */
+  public fun onBeforeCreateUi(screen: Screen, context: CircuitContext) {}
+
+  /**
+   * Called just after creating a [screenUi] for a given [screen]. The ui may be null if none was
+   * found.
+   */
+  public fun onAfterCreateUi(screen: Screen, screenUi: ScreenUi?, context: CircuitContext) {}
+
+  /** Called when no content was found and one or both of [presenter] and [screenUi] are null. */
+  public fun onUnavailableContent(
+    screen: Screen,
+    presenter: Presenter<*>?,
+    screenUi: ScreenUi?,
+    context: CircuitContext
+  ) {}
```

Comment:
It doesn't look like we call this anywhere

Reply:
good catch c47113c